iroh-resolver/src/unixfs_builder.rs

```diff
@@ -458,7 +459,8 @@ pub async fn add_file<S: Store>(store: Option<S>, path: &Path, wrap: bool) -> Re
     while let Some(part) = parts.next().await {
         let (cid, bytes) = part?;
         if let Some(ref store) = store {
-            store.put(cid, bytes, vec![]).await?;
+            let links = parse_links(&cid, &bytes)?;
+            store.put(cid, bytes, links).await?;
```

Comment:
we already know the links, when we encode things. instead of parsing them out, we should expose them from the underlying generator and use them here

Reply:
OK

Reply:
Digging into this now. This will be a bit of a bigger operation then... Got lots of places where the links need to be funneled though....